examples/Services/Planning/AddKeywordPlan.cs

```diff
@@ -0,0 +1,322 @@
+                CreateKeywordPlanKeywords(client, customerId, planAdGroupResource);
+                CreateKeywordPlanNegativeKeywords(client, customerId, planCampaignResource);
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+            }
+        }
+
+        /// <summary>
+        /// Creates the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <returns>The newly created keyword plan resource.</returns>
+        private string CreateKeywordPlan(GoogleAdsClient client, long customerId)
+        {
+            // Get the KeywordPlanService.
+            KeywordPlanServiceClient serviceClient = client.GetService(
+                Services.V0.KeywordPlanService);
+
+            // Create a keyword plan for next quarter forecast.
+            KeywordPlan keywordPlan = new KeywordPlan()
+            {
+                Name = "Keyword plan for traffic estimate #" + ExampleUtilities.GetRandomString(),
+                ForecastPeriod = new KeywordPlanForecastPeriod()
+                {
+                    DateInterval = KeywordPlanForecastInterval.NextQuarter
+                }
+            };
+
+            KeywordPlanOperation operation = new KeywordPlanOperation()
+            {
+                Create = keywordPlan
+            };
+
+            // Add the keyword plan.
+            MutateKeywordPlansResponse response = serviceClient.MutateKeywordPlans(
+                customerId.ToString(), new KeywordPlanOperation[] { operation });
+
+            // Display the results.
+            String planResource = response.Results[0].ResourceName;
+            Console.WriteLine($"Created keyword plan: {planResource}.");
+            return planResource;
+        }
+
+        /// <summary>
+        /// Creates the campaign for the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="keywordPlanResource">The keyword plan resource.</param>
+        /// <returns>The newly created campaign resource.</returns>
+        private string CreateKeywordPlanCampaign(GoogleAdsClient client, long customerId,
+            String keywordPlanResource)
+        {
+            // Get the KeywordPlanCampaignService.
+            KeywordPlanCampaignServiceClient serviceClient = client.GetService(
+                Services.V0.KeywordPlanCampaignService);
+
+            // Create a keyword plan campaign.
+            KeywordPlanCampaign campaign = new KeywordPlanCampaign()
```

Comment:
If a user wants to use an existing campaign, does that mean they have to query it and then input the corresponding parameters to create KeywordPlanCampaign? I am just wondering why we are not taking the campaign resource directly, but I can imagine the keyword planner may not take every campaign attribute.

Reply:
At the moment, they need to query the desired campaign and input the parameters.